### PR TITLE
feat: Add event subscription option to webhooks

### PR DESCRIPTION
**Pull request by pranavrajs**
- Added an attribute in the webhook object called `subscriptions`, and added the currently available events as default. 
- Removed `conversation_resolved` and `conversation_opened` events, we had this event deprecated in the documentation already.
- Updated the dashboard UI to allow selecting the events.
- Cleanup the frontend code.

Loom video:  https://www.loom.com/share/f58f4911cc9f497daa8cdd8b0cf50f62

#### List:

Screenshot 2022-04-24 at 4 04 47 PM

#### New Webhook:
Screenshot 2022-04-24 at 4 50 11 PM

Fixes https://github.com/chatwoot/chatwoot/issues/3536
Fixes https://github.com/chatwoot/chatwoot/issues/4539

### CODE REVIEW

**Review by iamsivin — COMMENTED** [review]
Added a small comment. Code LGTM ⚡️

File: app/javascript/dashboard/routes/dashboard/settings/integrations/Webhooks/WebhookRow.vue [hunk 1]
```diff
@@ -0,0 +1,83 @@
+<template>
+  <tr>
+    <td>
+      <div class="webhook--link">{{ webhook.url }}</div>
+      <span class="webhook--subscribed-events">
+        <span class="webhook--subscribed-label">
+          {{ $t('INTEGRATION_SETTINGS.WEBHOOK.SUBSCRIBED_EVENTS') }}:
+        </span>
+        <show-more :text="subscribedEvents" limit="60" />
```
Comment (iamsivin) [inline app/javascript/dashboard/routes/dashboard/settings/integrations/Webhooks/WebhookRow.vue:9]: The `limit` prop is expected as a number
```suggestion
        <show-more :text="subscribedEvents" :limit="60" />
```

**Review by iamsivin — DISMISSED** [review]
FE Code LGTM💯

File: app/javascript/dashboard/routes/dashboard/settings/integrations/Webhooks/NewWebHook.vue [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+      :header-content="
+        useInstallationName(
+          $t('INTEGRATION_SETTINGS.WEBHOOK.FORM.DESC'),
+          globalConfig.installationName
+        )
+      "
+    />
+    <webhook-form
+      :is-submitting="uiFlags.creatingItem"
+      :submit-label="$t('INTEGRATION_SETTINGS.WEBHOOK.FORM.EDIT_SUBMIT')"
```
Comment (muhsin-k) [inline app/javascript/dashboard/routes/dashboard/settings/integrations/Webhooks/NewWebHook.vue:14]:
```suggestion
      :submit-label="$t('INTEGRATION_SETTINGS.WEBHOOK.FORM.ADD_SUBMIT')"
```

File: app/javascript/dashboard/routes/dashboard/settings/integrations/Webhooks/webhookMixin.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+export default {
+  methods: {
+    getEventLabel(event) {
```
Comment (muhsin-k) [inline app/javascript/dashboard/routes/dashboard/settings/integrations/Webhooks/webhookMixin.js:3]: Add spec for `webhookMixin.js`

**Comment by muhsin-k** [comment]
QA has been done on local environment and all the events are working as expected

https://www.loom.com/share/9a5dffb6b0c44d2285ea06427872f3c6

**Review by nithindavid — COMMENTED** [review]
FE Part looks good

File: app/javascript/dashboard/routes/dashboard/settings/integrations/Webhooks/webhookMixin.js (hunk 3 above)
Comment (pranavrajs) [reply app/javascript/dashboard/routes/dashboard/settings/integrations/Webhooks/webhookMixin.js:3]: Done

**Review by sojan-official — APPROVED** [review]
LGTM

**Pull request merged**